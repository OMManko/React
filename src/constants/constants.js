export const endPoint = 'http://react-cdp-api.herokuapp.com/movies';

export const moviePosterSizes = {
    SMALL: 'moviePoster__small',
    MEDIUM: 'moviePoster__medium'
};

export const sortOptions = {
    RELEASE: 'release',
    RATING: 'vote_average'
};

export const filterOptions = {
    TITLE: 'title',
    GENRE: 'genres'
};


export const actionTypes = {
    SEARCH_BY: 'SEARCH_BY',
    SORT_BY: 'SORT_BY',
    SEARCH_MOVIES: 'SEARCH_MOVIES',
    UPDATE_INPUT_VALUE: 'UPDATE_INPUT_VALUE',
    SET_SELECTED_MOVIE: 'SET_SELECTED_MOVIE',
    RESET_SELECTED_MOVIE: 'RESET_SELECTED_MOVIE'
};

export const initialState = {
    moviesList: [],
    moviesFound: 0,
    searchInputValue: '',
    selectedSortOption: sortOptions.RELEASE,
    selectedFilterOption: filterOptions.TITLE,
    selectedMovieInfo: {}
};
