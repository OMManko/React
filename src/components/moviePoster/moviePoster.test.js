import { shallow } from 'enzyme';
import React from 'react';
import MoviePoster from './moviePoster';
import { moviePosterSizes } from "../../constants/constants";

describe('<MoviePoster />', () => {
    it('should render movie poster and match snapshot', () => {
        const component = shallow(<MoviePoster />);
        expect(component).toMatchSnapshot();
    });

    it('should set src to movie poster image', () => {
        const source = 'https://i.ytimg.com/vi/Uv50q2PudQw/maxresdefault.jpg';
        const component = shallow(<MoviePoster src={source}/>);
        expect(component).toMatchSnapshot();
        expect(component.find('img').props().src).toEqual(source);
    });

    it('should set size to movie poster', () => {
        const size = moviePosterSizes.MEDIUM;
        const component = shallow(<MoviePoster size={size}/>);
        expect(component).toMatchSnapshot();
        expect(component.props().className).toContain(size);
    });
});
