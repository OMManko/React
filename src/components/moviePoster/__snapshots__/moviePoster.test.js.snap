// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePoster /> should render movie poster and match snapshot 1`] = `ShallowWrapper {}`;

exports[`<MoviePoster /> should set size to movie poster 1`] = `ShallowWrapper {}`;

exports[`<MoviePoster /> should set src to movie poster image 1`] = `ShallowWrapper {}`;
